### PR TITLE
enable SSL via ENVs

### PR DESCRIPTION
**Pull request by anonym-HPI**
TODO:

- [x] disallow http traffic and redirect when ssl is enabled (maybe via ENV enabled by default)
- [x] ssl header hardening (disabling by default, probably but encouring it but warning, only enabling it after https works flawlessly even with docker restarts - after it http will not be possible on this domain for a year, without getting into every browser settings of every browser that visitted this domain)

needs https://github.com/hpi-sam/digital-fuesim-manv/pull/369

### CODE REVIEW

**Review by ClFeSc — CHANGES_REQUESTED** [review]

File: docker/Dockerfile
Comment (anonym-HPI) [reply docker/Dockerfile:62]: I have to, as the acme.sh script otherwise not works properly.

File: docker/cron-renew-session-ticket
Comment (ClFeSc) [reply docker/cron-renew-session-ticket]: Please make clear that this is an error in nginx, then I'm happy with it.

**Review by ClFeSc — COMMENTED** [review]
Just the two TODO comments remaining. Any feedback on them would be nice. Otherwise looks good. I haven't tested it, though.

**Comment by ClFeSc** [comment]
> Just the two TODO comments remaining. Any feedback on them would be nice. Otherwise looks good. I haven't tested it, though.

@anonym-HPI Still no updates?

**Comment by anonym-HPI** [comment]
@ClFeSc what about putting everything important in something like /dfm or /data or /persistent-data or ... ?

File: docker/Dockerfile
Comment (ClFeSc) [inline docker/Dockerfile:58]: Update comment with new env

File: docker/nginx/certs
Comment (ClFeSc) [inline docker/nginx/certs:1]: What is empty?

File: docker/Dockerfile
Comment (ClFeSc) [reply docker/Dockerfile]: @anonym-HPI Unfortunately, you've missed `DFM_` in the env name twice in this line.

Comment (ClFeSc) [reply docker/Dockerfile]: @anonym-HPI Unfortunately, now you've lost the `$`.

**Comment by anonym-HPI** [comment]
@ClFeSc waiting for final approval

**Review by ClFeSc — APPROVED** [review]

**Pull request merged**
